src/general/world_effects/patch_events/GlobalGlaciationEvent.cs

```diff
@@ -0,0 +1,284 @@
+
+        patch.Biome.ModifyLongTermCondition(Compound.Temperature, currentTemperature);
+        patch.Biome.ModifyLongTermCondition(Compound.Sunlight, currentSunlight);
+    }
+
+    /// <summary>
+    ///   Gets chunks from Iceshelf patch template and applies them to the patches.
+    /// </summary>
+    private void AddIceChunks(Patch patch)
+    {
+        var templateBiome = SimulationParameters.Instance.GetBiome(TemplateBiomeForIceChunks);
+        foreach (var configuration in IceChunksConfigurations)
+        {
+            var iceChunkConfiguration = templateBiome.Conditions.Chunks[configuration];
+            iceChunkConfiguration.Density *= 10;
+            patch.Biome.Chunks.Add(Prefix + configuration, iceChunkConfiguration);
+        }
+    }
+
+    private void LogEvent(Patch patch, double totalTimePassed)
+    {
+        patch.LogEvent(new LocalizedString("GLOBAL_GLACIATION_EVENT"),
+            true, true, "GlobalGlaciationEvent.svg");
+
+        if (patch.Visibility == MapElementVisibility.Shown)
+        {
+            targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_EVENT_LOG", patch.Name),
+                true, false, "GlobalGlaciationEvent.svg");
+        }
+
+        patch.AddPatchEventRecord(WorldEffectVisuals.GlobalGlaciation, totalTimePassed);
+    }
+
+    private void LogHeadUpEventWarning()
+    {
+        targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_EVENT_WARNING_LOG", generationsToTrigger),
+            true, true, "GlobalGlaciationEvent.svg");
+    }
+
+    private void LogBeginningOfGlaciation()
+    {
+        targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_START_EVENT_LOG"),
+            true, true, "GlobalGlaciationEvent.svg");
+    }
+
+    private void LogEndOfGlaciation()
+    {
+        targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_END_EVENT_LOG"),
+            true, true, "GlobalGlaciationEvent.svg");
+    }
+
+    private void FinishEvent()
+    {
+        hasEventAlreadyHappened = true;
+        foreach (var index in modifiedPatchesIds)
+        {
+            if (!targetWorld.Map.Patches.TryGetValue(index, out var patch))
+            {
+                GD.PrintErr("Patch exited the world");
+                continue;
+            }
+
+            PatchSnapshot patchSnapshot = patch.History[eventDuration];
+
+            ResetBackground(patch, patchSnapshot);
+            ResetEnvironment(patch, patchSnapshot);
+            RemoveChunks(patch);
+        }
+
+        LogEndOfGlaciation();
+    }
+
+    private void ResetBackground(Patch patch, PatchSnapshot patchSnapshot)
+    {
+        patch.CurrentSnapshot.Background = patchSnapshot.Background;
+    }
+
+    private void ResetEnvironment(Patch patch, PatchSnapshot patchSnapshot)
+    {
+        var previousTemperature = patchSnapshot.Biome.ChangeableCompounds[Compound.Temperature];
+        var previousSunlight = patchSnapshot.Biome.ChangeableCompounds[Compound.Sunlight];
```

Comment:
Shouldn't these read the long term here? If I recall correctly there's a method that clearly reads that. For example `TryGetCompound(Compound.Sunlight, CompoundAmountType.Biome, out var sunlight)`